File: library/src/main/java/com/zfs/commons/utils/PreferencesUtils.java
package com.zfs.commons.utils;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.text.TextUtils;

import com.zfs.commons.AppHolder;

/**
 * Created by zengfs on 2015/10/30.
 */
public class PreferencesUtils {

	private PreferencesUtils() {
		throw new AssertionError();
	}
	
	private static SharedPreferences getSharedPreferences(String name) {
	    if (TextUtils.isEmpty(name)) {
            return PreferenceManager.getDefaultSharedPreferences(AppHolder.getContext());
	    } else {
            return AppHolder.getContext().getSharedPreferences(name, 0);
	    }
    }
	
	public static void putString(String key, String value) {
        getSharedPreferences(null).edit().putString(key, value).apply();
	}

    public static void putStringImmediately(String key, String value) {
        getSharedPreferences(null).edit().putString(key, value).commit();
    }
    
	/**
	 * 失败返回默认值为null
	 */
	public static String getString(String key) {
		return getString(key, null);
	}

	public static String getString(String key, String defaultValue) {
		return getSharedPreferences(null).getString(key, defaultValue);
	}

	public static void putInt(String key, int value) {
        getSharedPreferences(null).edit().putInt(key, value).apply();
	}

    public static void putIntImmediately(String key, int value) {
        getSharedPreferences(null).edit().putInt(key, value).commit();
    }

	/**
	 * 失败返回默认值为-1
	 */
	public static int getInt(String key) {
		return getInt(key, -1);
	}

	public static int getInt(String key, int defaultValue) {
		return getSharedPreferences(null).getInt(key, defaultValue);
	}

	public static void putLong(String key, long value) {
        getSharedPreferences(null).edit().putLong(key, value).apply();
	}

    public static void putLongImmediately(String key, long value) {
        getSharedPreferences(null).edit().putLong(key, value).commit();
    }

	/**
	 * 失败返回默认值为-1
	 */
	public static long getLong(String key) {
		return getLong(key, -1L);
	}

	public static long getLong(String key, long defaultValue) {
		return getSharedPreferences(null).getLong(key, defaultValue);
	}

	public static void putFloat(String key, float value) {
        getSharedPreferences(null).edit().putFloat(key, value).apply();
	}

    public static void putFloatImmediately(String key, float value) {
        getSharedPreferences(null).edit().putFloat(key, value).commit();
    }

	/**
	 * 失败返回默认值为-1
	 */
	public static float getFloat(String key) {
		return getFloat(key, -1F);
	}

	public static float getFloat(String key, float defaultValue) {
		return getSharedPreferences(null).getFloat(key, defaultValue);
	}

	public static void putBoolean(String key, boolean value) {
        getSharedPreferences(null).edit().putBoolean(key, value).apply();
	}

    public static void putBooleanImmediately(String key, boolean value) {
        getSharedPreferences(null).edit().putBoolean(key, value).commit();
    }

	public static boolean getBoolean(String key) {
		return getBoolean(key, false);
	}

	public static boolean getBoolean(String key, boolean defaultValue) {
		return getSharedPreferences(null).getBoolean(key, defaultValue);
	}

	public static void removeKey(String key) {
        getSharedPreferences(null).edit().remove(key).apply();
	}
	
	public static void removeKeyImmediately(String key) {
        getSharedPreferences(null).edit().remove(key).commit();
	}

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putString(String name, String key, String value) {
        getSharedPreferences(name).edit().putString(key, value).apply();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putStringImmediately(String name, String key, String value) {
        getSharedPreferences(name).edit().putString(key, value).commit();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static String getString(String name, String key, String defaultValue) {
        return getSharedPreferences(name).getString(key, defaultValue);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putInt(String name, String key, int value) {
        getSharedPreferences(name).edit().putInt(key, value).apply();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putIntImmediately(String name, String key, int value) {
        getSharedPreferences(name).edit().putInt(key, value).commit();
    }

    /**
     * 失败返回默认值为-1
     */
    public static int getInt(String name, String key) {
        return getInt(name, key, -1);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static int getInt(String name, String key, int defaultValue) {
        return getSharedPreferences(name).getInt(key, defaultValue);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putLong(String name, String key, long value) {
        getSharedPreferences(name).edit().putLong(key, value).apply();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putLongImmediately(String name, String key, long value) {
        getSharedPreferences(name).edit().putLong(key, value).commit();
    }

    /**
     * 失败返回默认值为-1
     */
    public static long getLong(String name, String key) {
        return getLong(name, key, -1L);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static long getLong(String name, String key, long defaultValue) {
        return getSharedPreferences(name).getLong(key, defaultValue);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putFloat(String name, String key, float value) {
        getSharedPreferences(name).edit().putFloat(key, value).apply();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putFloatImmediately(String name, String key, float value) {
        getSharedPreferences(name).edit().putFloat(key, value).commit();
    }

    /**
     * 失败返回默认值为-1
     */
    public static float getFloat(String name, String key) {
        return getFloat(name, key, -1F);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static float getFloat(String name, String key, float defaultValue) {
        return getSharedPreferences(name).getFloat(key, defaultValue);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putBoolean(String name, String key, boolean value) {
        getSharedPreferences(name).edit().putBoolean(key, value).apply();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void putBooleanImmediately(String name, String key, boolean value) {
        getSharedPreferences(name).edit().putBoolean(key, value).commit();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static boolean getBoolean(String name, String key) {
        return getBoolean(name, key, false);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static boolean getBoolean(String name, String key, boolean defaultValue) {
        return getSharedPreferences(name).getBoolean(key, defaultValue);
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void removeKey(String name, String key) {
        getSharedPreferences(name).edit().remove(key).apply();
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    public static void removeKeyImmediately(String name, String key) {
        getSharedPreferences(name).edit().remove(key).commit();
    }
}

File: library/src/main/java/com/snail/commons/utils/PreferencesUtils.kt
package com.snail.commons.utils

import android.content.SharedPreferences
import android.preference.PreferenceManager
import android.text.TextUtils

import com.snail.commons.AppHolder

/**
 * Created by zengfs on 2015/10/30.
 */
@Deprecated(message = "建议使用MMKV，详情：https://github.com/Tencent/MMKV")
object PreferencesUtils {

    private fun getSharedPreferences(name: String?): SharedPreferences {
        return if (TextUtils.isEmpty(name)) {
            PreferenceManager.getDefaultSharedPreferences(AppHolder.context)
        } else {
            AppHolder.context.getSharedPreferences(name, 0)
        }
    }

    @JvmStatic 
    fun putString(key: String, value: String) {
        getSharedPreferences(null).edit().putString(key, value).apply()
    }

    @JvmStatic 
    fun putStringImmediately(key: String, value: String) {
        getSharedPreferences(null).edit().putString(key, value).commit()
    }

    @JvmStatic 
    fun getString(key: String, defaultValue: String): String {
        return getSharedPreferences(null).getString(key, defaultValue)
    }
    
    @JvmStatic 
    fun getString(key: String): String? {
        return getSharedPreferences(null).getString(key, null)
    }

    @JvmStatic 
    fun putInt(key: String, value: Int) {
        getSharedPreferences(null).edit().putInt(key, value).apply()
    }

    @JvmStatic 
    fun putIntImmediately(key: String, value: Int) {
        getSharedPreferences(null).edit().putInt(key, value).commit()
    }

    @JvmOverloads
    @JvmStatic 
    fun getInt(key: String, defaultValue: Int = -1): Int {
        return getSharedPreferences(null).getInt(key, defaultValue)
    }

    @JvmStatic 
    fun putLong(key: String, value: Long) {
        getSharedPreferences(null).edit().putLong(key, value).apply()
    }

    @JvmStatic 
    fun putLongImmediately(key: String, value: Long) {
        getSharedPreferences(null).edit().putLong(key, value).commit()
    }

    @JvmOverloads
    @JvmStatic 
    fun getLong(key: String, defaultValue: Long = -1L): Long {
        return getSharedPreferences(null).getLong(key, defaultValue)
    }

    @JvmStatic 
    fun putFloat(key: String, value: Float) {
        getSharedPreferences(null).edit().putFloat(key, value).apply()
    }

    @JvmStatic 
    fun putFloatImmediately(key: String, value: Float) {
        getSharedPreferences(null).edit().putFloat(key, value).commit()
    }

    @JvmOverloads
    @JvmStatic 
    fun getFloat(key: String, defaultValue: Float = -1f): Float {
        return getSharedPreferences(null).getFloat(key, defaultValue)
    }

    @JvmStatic 
    fun putBoolean(key: String, value: Boolean) {
        getSharedPreferences(null).edit().putBoolean(key, value).apply()
    }

    @JvmStatic 
    fun putBooleanImmediately(key: String, value: Boolean) {
        getSharedPreferences(null).edit().putBoolean(key, value).commit()
    }

    @JvmOverloads
    @JvmStatic 
    fun getBoolean(key: String, defaultValue: Boolean = false): Boolean {
        return getSharedPreferences(null).getBoolean(key, defaultValue)
    }

    @JvmStatic 
    fun removeKey(key: String) {
        getSharedPreferences(null).edit().remove(key).apply()
    }

    @JvmStatic 
    fun removeKeyImmediately(key: String) {
        getSharedPreferences(null).edit().remove(key).commit()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putString(name: String, key: String, value: String) {
        getSharedPreferences(name).edit().putString(key, value).apply()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putStringImmediately(name: String, key: String, value: String) {
        getSharedPreferences(name).edit().putString(key, value).commit()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun getString(name: String, key: String, defaultValue: String): String? {
        return getSharedPreferences(name).getString(key, defaultValue)
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putInt(name: String, key: String, value: Int) {
        getSharedPreferences(name).edit().putInt(key, value).apply()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putIntImmediately(name: String, key: String, value: Int) {
        getSharedPreferences(name).edit().putInt(key, value).commit()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmOverloads
    @JvmStatic 
    fun getInt(name: String, key: String, defaultValue: Int = -1): Int {
        return getSharedPreferences(name).getInt(key, defaultValue)
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putLong(name: String, key: String, value: Long) {
        getSharedPreferences(name).edit().putLong(key, value).apply()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putLongImmediately(name: String, key: String, value: Long) {
        getSharedPreferences(name).edit().putLong(key, value).commit()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmOverloads
    @JvmStatic 
    fun getLong(name: String, key: String, defaultValue: Long = -1L): Long {
        return getSharedPreferences(name).getLong(key, defaultValue)
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putFloat(name: String, key: String, value: Float) {
        getSharedPreferences(name).edit().putFloat(key, value).apply()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putFloatImmediately(name: String, key: String, value: Float) {
        getSharedPreferences(name).edit().putFloat(key, value).commit()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmOverloads
    @JvmStatic 
    fun getFloat(name: String, key: String, defaultValue: Float = -1f): Float {
        return getSharedPreferences(name).getFloat(key, defaultValue)
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putBoolean(name: String, key: String, value: Boolean) {
        getSharedPreferences(name).edit().putBoolean(key, value).apply()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun putBooleanImmediately(name: String, key: String, value: Boolean) {
        getSharedPreferences(name).edit().putBoolean(key, value).commit()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmOverloads
    @JvmStatic 
    fun getBoolean(name: String, key: String, defaultValue: Boolean = false): Boolean {
        return getSharedPreferences(name).getBoolean(key, defaultValue)
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun removeKey(name: String, key: String) {
        getSharedPreferences(name).edit().remove(key).apply()
    }

    /**
     * 使用指定的SharedPreferences
     * @param name SharedPreferences的名
     */
    @JvmStatic 
    fun removeKeyImmediately(name: String, key: String) {
        getSharedPreferences(name).edit().remove(key).commit()
    }
}